pkg/controller/core/workload_controller.go

```diff
@@ -137,8 +137,9 @@ func (r *WorkloadReconciler) Reconcile(ctx context.Context, req ctrl.Request) (c
 		return ctrl.Result{}, nil
 	}
 
-	if rejectedChecks := workload.GetRejectedChecks(&wl); len(rejectedChecks) > 0 {
-		// Finish the workload
+	// If the workload has rejected admission checks and is not admitted mark it as finished.
+	// If it has rejected admission checks and is admitted, continue the reconcile in  order to trigger its eviction.
+	if rejectedChecks := workload.GetRejectedChecks(&wl); len(rejectedChecks) > 0 && !workload.IsAdmitted(&wl) {
```

Comment:
Can we have an integration test for the fixed scenario?

Reply:
or at least unit, but there should be coverage.